File: ruby-sdk/lib/cleanster/api/webhooks_api.rb
module Cleanster
  module Api
    # WebhooksApi — manage real-time event notification endpoints.
    class WebhooksApi
      def initialize(http)
        @http = http
      end

      # Return all configured webhook endpoints.
      # @return [Models::ApiResponse]
      def list_webhooks
        raw = @http.get("/v1/webhooks")
        Models::ApiResponse.from_hash(raw)
      end

      # Register a new webhook endpoint.
      #
      # @param request [Hash] url (HTTPS endpoint), event, and optional extra fields.
      # @return [Models::ApiResponse]
      def create_webhook(request)
        raw = @http.post("/v1/webhooks", body: request)
        Models::ApiResponse.from_hash(raw)
      end

      # Update an existing webhook endpoint configuration.
      #
      # @param webhook_id [Integer]
      # @param request    [Hash] Updated fields.
      # @return [Models::ApiResponse]
      def update_webhook(webhook_id, request)
        raw = @http.put("/v1/webhooks/#{webhook_id}", body: request)
        Models::ApiResponse.from_hash(raw)
      end

      # Delete a webhook endpoint.
      #
      # @param webhook_id [Integer]
      # @return [Models::ApiResponse]
      def delete_webhook(webhook_id)
        raw = @http.delete("/v1/webhooks/#{webhook_id}")
        Models::ApiResponse.from_hash(raw)
      end
    end
  end
end

File: php-sdk/src/Api/WebhooksApi.php
<?php

declare(strict_types=1);

namespace Cleanster\Api;

use Cleanster\HttpClient;
use Cleanster\Models\ApiResponse;

/**
 * Manages real-time event notification endpoints.
 */
final class WebhooksApi
{
    public function __construct(private readonly HttpClient $http) {}

    /** Return all configured webhook endpoints. */
    public function listWebhooks(): ApiResponse
    {
        $raw = $this->http->get('/v1/webhooks');
        return new ApiResponse($raw['status'] ?? 200, $raw['message'] ?? 'OK', $raw['data'] ?? []);
    }

    /**
     * Register a new webhook endpoint for booking event notifications.
     *
     * @param string $url   Your HTTPS endpoint URL.
     * @param string $event Event type (e.g., "booking.status_changed").
     */
    public function createWebhook(string $url, string $event): ApiResponse
    {
        $raw = $this->http->post('/v1/webhooks', ['url' => $url, 'event' => $event]);
        return new ApiResponse($raw['status'] ?? 200, $raw['message'] ?? 'OK', $raw['data'] ?? []);
    }

    /**
     * Update the URL or event type of an existing webhook.
     *
     * @param int    $webhookId Webhook to update.
     * @param string $url       New endpoint URL.
     * @param string $event     New event type.
     */
    public function updateWebhook(int $webhookId, string $url, string $event): ApiResponse
    {
        $raw = $this->http->put("/v1/webhooks/{$webhookId}", ['url' => $url, 'event' => $event]);
        return new ApiResponse($raw['status'] ?? 200, $raw['message'] ?? 'OK', $raw['data'] ?? []);
    }

    /** Remove a webhook endpoint. */
    public function deleteWebhook(int $webhookId): ApiResponse
    {
        $raw = $this->http->delete("/v1/webhooks/{$webhookId}");
        return new ApiResponse($raw['status'] ?? 200, $raw['message'] ?? 'OK', $raw['data'] ?? []);
    }
}
